pkg/operator/ceph/controller/spec.go

```diff
@@ -570,12 +570,18 @@ func GenerateLivenessProbeExecDaemon(daemonType, daemonID string) *v1.Probe {
 				//
 				// Example:
 				// env -i sh -c "ceph --admin-daemon /run/ceph/ceph-osd.0.asok status"
+				//
+				// Ceph gives pretty un-diagnostic error message when `ceph status` or `ceph mon_status` command fails.
+				// We'd better ignore this message and give users more simple one.
+				// The workaround below is bit dirty; but it works.
+				// ref: https://github.com/rook/rook/issues/9846
 				Command: []string{
 					"env",
 					"-i",
 					"sh",
 					"-c",
-					fmt.Sprintf("ceph --admin-daemon %s %s", confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand()),
+					fmt.Sprintf("ceph --admin-daemon %s %s 2>/dev/null; if [ \\$? -ne 0 ]; then echo %s; exit 1; fi",
+						confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand(), "ceph daemon health check failed."),
```

Comment:
Also a small nit that Rook follows the convention that error messages don't end in periods.

```suggestion
						confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand(), "ceph daemon health check failed"),
```